### PR TITLE
Phone sign in and verification web captcha

### PR DESCRIPTION
**Pull request by alexis-kochetov**

### CODE REVIEW

File: src/dotnet/UI.Blazor/Components/SignIn/Modal/ProviderSelectStep.razor [hunk 1]
```diff
@@ -53,30 +57,66 @@
                 <InputText
                     id="@_stepModel.PhoneFieldId"
                     @bind-Value="_stepModel.Phone"
+                    disabled="@(codeHasBeenSent)"
```
Comment (crui3er) [inline src/dotnet/UI.Blazor/Components/SignIn/Modal/ProviderSelectStep.razor]: brackets are excessive

File: src/dotnet/UI.Blazor/Module/BlazorUICoreModule.cs [hunk 2]
```diff
@@ -107,6 +108,7 @@ protected override void InjectServices(IServiceCollection services)
         services.AddScoped(c => new ShareUI(c.UIHub()));
         services.AddScoped(_ => new ToastUI());
         fusion.AddService<LiveTime>(ServiceLifetime.Scoped);
+        fusion.AddService<TotpUI>(ServiceLifetime.Scoped);
```
Comment (crui3er) [inline src/dotnet/UI.Blazor/Module/BlazorUICoreModule.cs:111]: We usually register services with implementation factory.

File: src/dotnet/UI.Blazor/Services/CaptchaUI/CaptchaUI.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    private readonly DotNetObjectReference<ICaptchaUIBackend> _blazorRef;
+    private UIHub Hub { get; }
+
+    public CaptchaUI(UIHub hub)
+    {
+        Hub = hub;
+        _blazorRef = DotNetObjectReference.Create<ICaptchaUIBackend>(this);
+    }
+
+    public bool IsConfigured => !string.IsNullOrWhiteSpace(SiteKey);
```
Comment (crui3er) [inline src/dotnet/UI.Blazor/Services/CaptchaUI/CaptchaUI.cs]: Put properties above methods/constructors.

File: src/dotnet/Users.Service/Captcha.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        catch (RpcException ex) when (ex.StatusCode == StatusCode.PermissionDenied) {
+            return new RecaptchaValidationResult(false, "reCAPTCHA is not configured.");
+        }
+        catch (Exception e) {
+            Log.LogWarning(e, "Error validating reCAPTCHA token");
+            return new RecaptchaValidationResult(false, "reCAPTCHA validation failed.");
+        }
+    }
+
+    public void Dispose()
```
Comment (crui3er) [inline src/dotnet/Users.Service/Captcha.cs]: I nothing to do in Dispose we can remove IDisposable interface

**Review by crui3er — APPROVED** [review]

File: src/dotnet/UI.Blazor/Module/BlazorUICoreModule.cs (hunk 2 above)
Comment (alexis-kochetov) [reply src/dotnet/UI.Blazor/Module/BlazorUICoreModule.cs:111]: I'm not sure that we do so for Fusion services

Comment (crui3er) [reply src/dotnet/UI.Blazor/Module/BlazorUICoreModule.cs:111]: You are right. It register proxy class internally. Did not put attention that AddService is called for fusion.

File: src/dotnet/UI.Blazor/Components/Phone/PhoneVerifier.razor [hunk 5]
```diff
@@ -2,8 +2,24 @@
...
+    var buttonText = _validationState switch {
+        ValidationState.Validating => "Validating...",
+        ValidationState.Invalid => "Unavailable",
+        ValidationState.CodeSentRecently => "Code sent",
+        _ => "Verify by SMS",
+    };
+    var description = _validationState switch {
+        ValidationState.Invalid => "Please try again later.",
+        ValidationState.CodeSentRecently => "You can try again in a few minutes.",
+        _ => "Verify your phone number to be able to sign in with it.",
```
Comment (alexyakunin) [inline src/dotnet/UI.Blazor/Components/Phone/PhoneVerifier.razor]: Verify your phone number to sign in to Actual Chat.

File: src/dotnet/UI.Blazor/Module/BlazorUICoreModule.cs (hunk 2 above)
Comment (alexyakunin) [reply src/dotnet/UI.Blazor/Module/BlazorUICoreModule.cs:111]: Yep, Fusion services activate some proxy rather than a service.

**Review by alexyakunin — CHANGES_REQUESTED** [review]
One text edit + minor stuff.

**Pull request merged**